File: frontend/src/shared/lib/attachment-limits.ts
/**
 * Centralized attachment limit constants.
 *
 * Keep in sync with backend: AttachmentLimits.java
 * (com.wellkorea.backend.shared.constant.AttachmentLimits)
 */

export const ATTACHMENT_LIMITS = {
  /** Maximum individual file size (50MB) */
  MAX_FILE_SIZE: 50 * 1024 * 1024,

  /** Maximum total attachment size per entity (20MB) */
  MAX_TOTAL_SIZE: 20 * 1024 * 1024,

  /** Maximum number of attachments per entity */
  MAX_ATTACHMENT_COUNT: 20,
} as const;

/**
 * Format bytes to human-readable size (e.g., "2.5 MB").
 */
export function formatFileSize(bytes: number): string {
  const units = ['B', 'KB', 'MB', 'GB'];
  let unitIndex = 0;
  let size = bytes;

  while (size >= 1024 && unitIndex < units.length - 1) {
    size /= 1024;
    unitIndex++;
  }

  return `${size.toFixed(size >= 10 ? 0 : 1)} ${units[unitIndex]}`;
}

File: backend/src/main/java/com/wellkorea/backend/supporting/storage/domain/constant/AttachmentLimits.java
package com.wellkorea.backend.supporting.storage.domain.constant;

/**
 * Centralized attachment limit constants.
 * <p>
 * Single source of truth for file size and count limits across the application.
 * Used by AttachmentReference, ServicePurchaseRequest, and RfqEmailService.
 * <p>
 * Frontend equivalent: {@code shared/lib/attachment-limits.ts}
 */
public final class AttachmentLimits {

    private AttachmentLimits() {
        // Prevent instantiation
    }

    /**
     * Maximum individual file size (50MB).
     * Same limit as BlueprintAttachment uploads.
     */
    public static final long MAX_FILE_SIZE = 52_428_800L;

    /**
     * Maximum total attachment size per entity (20MB).
     * Applies to ServicePurchaseRequest total attachments.
     * Also used as email attachment size limit.
     */
    public static final long MAX_TOTAL_SIZE = 20 * 1024 * 1024L;

    /**
     * Maximum number of attachments per entity.
     */
    public static final int MAX_ATTACHMENT_COUNT = 20;
}
